egs/librispeech/ASR/transducer_stateless/alignment.py

```diff
@@ -0,0 +1,264 @@
+# Copyright    2022  Xiaomi Corp.        (authors: Fangjun Kuang)
+#
+# See ../../../../LICENSE for clarification regarding multiple authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from dataclasses import dataclass
+from typing import Iterator, List, Optional
+
+import sentencepiece as spm
+import torch
+from model import Transducer
+
+# The force alignment problem can be formulated as finding
+# a path in a rectangular lattice, where the path starts
+# from the lower left corner and ends at the upper right
+# corner. The horizontal axis of the lattice is `t` (representing
+# acoustic frame indexes) and the vertical axis is `u` (representing
+# BPE tokens of the transcript).
+#
+# Beam search is used to find the path with the
+# highest log probabilities.
+#
+# It assumes the maximum number of symbols that can be
+# emitted per frame is 1. You can use `--modified-transducer-prob`
+# from `./train.py` to train a model that satisfies this assumption.
+
+
+# AlignItem is a node in the lattice, where its
+# len(ys) equals to `t` and pos_u is the u coordinate
+# in the lattice.
```

Comment:
people may not know what the u co-ordinate refers to here (actually, I don't).  Is it the word position?

Reply:
In transducer training, the output shape is usually denoted as (N, T, U, V), instead of `(N, S, T, V)` used in k2, I think.
The notation `t` and `u` is from https://arxiv.org/pdf/1211.3711.pdf?ref=hackernoon.com
<img width="915" alt="Screen Shot 2022-03-07 at 5 21 09 PM" src="https://user-images.githubusercontent.com/5284924/157003060-9cdd33b9-6170-436d-8559-bf9890414258.png">

Since our modelling units are BPE tokens, so `u` actually denotes token position.

Reply:
OK, please clarify all this in the docs.